src/sg_expr.rs

```diff
@@ -230,7 +230,7 @@ impl Formattable for &Expr {
                     if let Some(c) = e.capture {
                         append_comments(out, base_indent, &mut sg, c.span.start());
                         sg.seg_unsplit(out, " ");
```

Comment:
Can you delete this line too? Otherwise looks good.

For background, basically `seg_unsplit` means add a space here as long as the "split group" isn't split.  Typically this would be used for things like `a, b` where there's a space after the comma, but when the element is split the `a,` and `b` would be on different lines and the space isn't necessary.

In this case `async move` is on the same line even when splitting (too little gains) but the space still disappears since it's `unsplit`, which is incorrect.  This was a bad translation of the code I moved away from.

Reply:
which line? 235?

Reply:
232, this `seg_unsplit(out, " ")` - it'll be redundant with your fix.